server/konnectors/linkedin.coffee

```diff
@@ -0,0 +1,328 @@
+
+
+module.exports =
+
+    name: "Linkedin"
+    slug: "linkedin"
+    description: "konnector description linkedin"
+
+    fields:
+        login: "text"
+        password: "password"
+    models:
+        contact: Contact
+
+    fetch: (requiredFields, callback) ->
+        log.info 'Import started'
+        fetcher.new()
+            .use(retrieveTokens)
+            .use(logIn)
+            .use(retrieveListContact)
+            .use(prepareCozyContacts)
+            .use(retrieveContacts)
+            .args(requiredFields, {}, {})
+            .fetch (err, fields, entries) ->
+                return callback err if err
+                log.info "Import finished"
+                callback()
+
+
+retrieveTokens = (requiredFields, entries, data, next) ->
+    log.info 'Retrieve Tokens'
+    opts =
+        url: 'https://linkedin.com'
+        jar: true
+        headers:
+            'User-Agent': agent
+
+    request.get opts, (err, res, body) ->
+        return next err if err
+
+        if body.status? and body.status == 'error'
+            next new Error(body.status_details)
+
+        $ = cheerio.load body
+        entries.csrfToken = $('#loginCsrfParam-login').val()
+        entries.sourceAlias = $('#sourceAlias-login').val()
+        entries.accountName = requiredFields.login
+        log.info 'Tokens retrieved'
+        next()
+
+
+
+logIn = (requiredFields, entries, data, next) ->
+    log.info 'Attempt login'
+    opts =
+        url: 'https://www.linkedin.com/uas/login-submit'
+        jar: true
+        form:
+            session_key: requiredFields.login
+            session_password: requiredFields.password
+            isJsEnabled: "false"
+            loginCsrfParam: entries.csrfToken
+            sourceAlias: entries.sourceAlias
+            submit: "Sign+in"
+            headers:
+                'User-Agent': agent
+
+    request.post opts, (err, res, body) ->
+        return next err if err
+
+        if body == ""
+            log.info 'Login success'
+            next()
+        else
+            log.error 'Bad login / password'
+            next(new Error("Bad login/Password"))
+
+
+retrieveListContact = (requiredFields, entries, data, next) ->
+    log.info 'Retrieve list contact'
+
+    contacts = "https://www.linkedin.com/contacts/api/contacts/full/?start=0&count=10000&fields=name"
+    opts =
+        url: contacts
+        jar: true
+        json: true
+        headers:
+            'User-Agent': agent
+
+    request.get opts, (err, res, body) ->
+        return next err if err
+        if body.status? and body.status == 'error'
+            next new Error(body.status_details)
+        entries.listContacts = body.contacts
+        if not entries.listContacts?
+            next new Error("Error retrieving contacts from request")
+        log.info 'List contact OK'
+        next()
+
+
+# Same as Google Konnector function
+prepareCozyContacts = (requiredFields, entries, data, next) ->
```

Comment:
Can you add to this function a comment explaining why and how you retrieve Cozy contacts?